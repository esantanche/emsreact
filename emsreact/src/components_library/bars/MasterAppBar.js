import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import AppBar from '@material-ui/core/AppBar';

// Material UI uses withStyles to inject an array of styles into the DOM as CSS, using JSS
import { withStyles } from "@material-ui/core/styles/index";

// This array of style gets injected into the component as "classes"
const styles = {
    MasterAppBar: {
        borderRadius: "4px",
    }
};

/**
 * The AppBar in the header. Its children are the home button and the menu button.
 *
 * @param {object} classes Classes injected by withStyles function
 */
class MasterAppBar extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired
    };

    render() {

        const { classes } = this.props;

        return (

            <AppBar position="sticky" className={classes.MasterAppBar}>

                {this.props.children}

            </AppBar>

        )
    }

}

export default withStyles(styles)(MasterAppBar);

