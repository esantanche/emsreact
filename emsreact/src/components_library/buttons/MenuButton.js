import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Button from "@material-ui/core/Button";

// Material UI icons
import MenuIcon from "@material-ui/icons/Menu";

// Material UI functions that make classes available
import { withStyles } from "@material-ui/core/styles/index";

const styles = {
    button: {
        float: "right"
    }
};

/**
 * This is the menu button, the one made of three horizontal lines
 *
 * Props
 * @param {function} onclick Function to call when the user clicks on the menu button
 */
class MenuButton extends Component {

    static propTypes = {
        onclick: PropTypes.func
    };

    render() {

        const { onclick, classes } = this.props;

        return (

            <Button onClick={onclick}
                    className={classes.button}>
                <MenuIcon />
            </Button>

        );
    }
}

export default withStyles(styles)(MenuButton);