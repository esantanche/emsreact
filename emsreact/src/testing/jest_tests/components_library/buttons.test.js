// FIXME docs

// import ReactDOM from "react-dom";
import React from "react";
import renderer from 'react-test-renderer';
//import MasterAvatar from "../../../components_library/avatars/MasterAvatar";

import red from "@material-ui/core/colors/red";
import MasterAppBar from "../../../components_library/bars/MasterAppBar";
import MenuButton from "../../../components_library/buttons/MenuButton";

import { shallow } from 'enzyme';

// FIXME evertyhing

// FIXME Here we test components in components_library/avatars

it('MenuButton renders correctly', () => {
    const tree = renderer
        .create(<MenuButton />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('MenuButton, test click event', () => {
    const mockCallBack = jest.fn();

    const button = shallow(<MenuButton onClick={mockCallBack} />);
    button.find('MenuButton').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
});