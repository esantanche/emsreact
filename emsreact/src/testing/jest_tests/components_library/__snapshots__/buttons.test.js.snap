// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton renders correctly 1`] = `
<button
  className="MuiButtonBase-root-26 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7 MenuButton-button-1"
  disabled={false}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-29"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </g>
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-36"
  />
</button>
`;
